CefSharp.Core/Internals/Serialization/V8Serialization.cpp

```diff
@@ -111,6 +111,18 @@ namespace CefSharp
                     }
                     list->SetList(index, subList);
                 }
+                // Serialize dictionary to CefDictionary (key,value pairs)
+                else if (type->IsGenericType && type->GenericTypeArguments->Length == 2 && type->Name->StartsWith("Dict"))
```

Comment:
Can just check for dictionary using https://msdn.microsoft.com/en-us/library/system.type.isassignablefrom(v=vs.110).aspx

Reply:
If the user returns an object of Dictionary with more specific parameters, for example, `Dictionary<string, string>` instead of `Dictionary<string, object>`, `IsAssignableFrom` returns false:

```
typeof(Dictionary<string, object>).IsAssignableFrom(typeof(Dictionary<string, string>)) === false
````

Reply:
A chick for just straight `IDictionary` should be sufficient, not need to limit things to the generic version.

Reply:
You are later casting to `IDitionary`, so the generic checks aren't required. And using `Convert.ToString` should gracefully handle the type conversion of the key .

https://msdn.microsoft.com/en-us/library/system.convert.tostring(v=vs.110).aspx

Reply:
Don't have any experience with managed C++ and get confused with C#.... Will make a commit later

Reply:
It will be something like `type->IsAssignableFrom(System::Collections::IDictionary::typeid)`